### PR TITLE
[KED - 3083] Header component

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

resolves - https://github.com/kedro-org/kedro-viz/issues/762

This PR is to set up the Header section as set out by the Kedro website design.

## Development notes

There is a slight confusion regarding the margins for medium breakpoint. Susanna has implemented the change in Case Study PR. I will update the ticket once that change has been approved and merged. For now for medium breakpoint, the margins are smaller

## QA notes
Please test the performance according to different break points. I will also sync up with design to go through the details of this PR.

## Checklist
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Added tests to cover my changes

### CODE REVIEW

File: components/header/header.module.scss
Comment (tynandebold) [inline components/header/header.module.scss:36]: If you're setting something to 0 you generally don't need a unit.

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:2]: I've been trying to put the style import as the last thing at the top of each file, [like here](https://github.com/kedro-org/kedro-website/blob/main/components/why-kedro/why-kedro.tsx), to try and separate concerns a little bit at the top (e.g. all JS imports and then styles below that).

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:11]: Try and put a space before `return` statements.

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:7]: I don't think we need this `assets` folder. We could just go `/public/icons/filename`.

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:24]: I think this should be `Why Kedro?` with a question mark.

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:40]: I believe our pattern is that these external links should open in a [new tab](https://github.com/kedro-org/kedro-website/blob/main/components/hero/hero.tsx#L19-L20).

File: public/assets/icons/burger.tsx
Comment (tynandebold) [inline public/assets/icons/burger.tsx:5]: It's nice you've created created components for the SVGs. I'm curious why you took this approach instead of treating them like `jpgs` or `pngs` though?

I was saying the same thing to Susanna yesterday, and that was that I think we can just have `.svg` files and reference them in the Next image component, with the source path being e.g.  `/public/icon/burger.svg`.

Let me know what you think. If you go this route you'll have to modify the styles a little bit, but it shouldn't be much.

File: components/header/header.module.scss
Comment (tynandebold) [inline components/header/header.module.scss:73]: Should this just be 0? When it's the height of the header it pushes down the content on mobile and therefore doesn't look centered.

File: styles/global.scss
Comment (tynandebold) [inline styles/global.scss:36]: What this for? I'm not sure I understand.

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:17]: Should this be a link to just `/`, a.k.a. a `home` link?

**Review by tynandebold — CHANGES_REQUESTED** [review]
Looking great so far! It's really coming together now that we have a header in there.

I've left some comments, though nothing huge. Two call outs from my comments: I don't think we need an `assets` folder within the `public` folder, and I think we can just use svg files for logos. Let me know if that or anything is unclear and if you have questions.

And two other things: we definitely need some more space between the header and the hero below. This is too tight, and the red squiggle is where more space is needed: 

Screen Shot 2022-03-08 at 09 37 21

Would you mind adding in some space there? Check the design for how much.

And lastly, if you check the browser console you'll see there's an error trying to load the poster for the video. I think you just need to update the path.

File: components/header/header.module.scss
Comment (studioswong) [inline components/header/header.module.scss:65]: Let's change this to `rem` for consistency with all other units in this style document

File: components/header/header.module.scss
Comment (studioswong) [reply components/header/header.module.scss:23]: agree, the unit should be `rem`

**Review by studioswong — CHANGES_REQUESTED** [review]
All in all, nice work setting this up. 

The header looks good on desktop, though I agree with Tynan that we need to add more space between the header and the hero section. 

I do notice though that there are discrepancies with this execution and the design for the mobile version ( see below), mainly on the position of the text, the discord and github logos ( I think only the logos will be shown without text), as well as the background arrangement ( the background for the text dropdown is a solid color, with the non-text area being a partially transparent overlay; currently the entire background is a partially transparent overlay). 

image

image

image

Another thing to note is that the buttons to external links opens up on the same tab, but with other sections we have established a pattern of opening in a separate tab - would be great if you can update the `target` attribute for those links.

File: public/assets/icons/burger.tsx
Comment (rashidakanchwala) [reply public/assets/icons/burger.tsx:5]: I did it because the tests were failing. Do we need to mock image set-up? and i saw Susanna did it, so yeah followed.

File: styles/global.scss
Comment (rashidakanchwala) [reply styles/global.scss:36]: when you navigate from one section to another, the section is like right on the top. The title gets hidden behind the header, so that's why i put this margin.

File: public/assets/icons/burger.tsx
Comment (tynandebold) [reply public/assets/icons/burger.tsx:5]: Do you remember what the errors were saying?

File: styles/global.scss
Comment (tynandebold) [reply styles/global.scss:36]: Oh nice!

File: components/header/header.module.scss
Comment (rashidakanchwala) [reply components/header/header.module.scss:73]: It is intentional to push down the content. According to Mackay's new designs that I shared on slack. On mobile, the user uses his/her thumb, so there shld be more space between header and menu options. Let me know if u still find it  slightly different from design

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:38]: I think we have the link for this. You can use: https://kedro.readthedocs.io/en/stable/

File: components/header/header.module.scss
Comment (tynandebold) [inline components/header/header.module.scss:33]: Try and have a space after the `:` here. This is something that should be picked up automatically by prettier when you save the file but also when things are committed or pushed. Maybe that isn't working as it should. I'll investigate.

File: components/header/header.module.scss
Comment (tynandebold) [inline components/header/header.module.scss:71]: We don't need browser prefixes as we're only supporting modern browsers and back a couple of versions :)

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:16]: If you feel like, we can remove this `layout="intrinsic"` prop since it's the default, though it's no problem to have it. Your call.

If you remove, you can do so from all other places in this file.

**Review by tynandebold — APPROVED** [review]
Great job, thanks so much for bringing this home.

I left a couple small comments. None of them are blocking, though please try and get them in before merging.

Lastly, I see a `.DS_Store` file being added in the root and `public` directly but that should be ignored by our `.gitignore`. Not sure why it isn't. Maybe look into that and also ensure they aren't added to the PR?

File: components/header/header.module.scss
Comment (rashidakanchwala) [reply components/header/header.module.scss:33]: this is because i tried to resolve conflicts on github. i am fixing it now

File: components/header/header.tsx
Comment (tynandebold) [inline components/header/header.tsx:37]: Don't forget to have this open in a new tab since it's an external link!

File: components/header/header.tsx
Comment (studioswong) [inline components/header/header.tsx:21]: This can be deleted. ( P.S are the links updated?)

File: components/hero/hero.module.scss
Comment (studioswong) [inline components/hero/hero.module.scss:8]: Please state in rem for consistency

File: components/header/header.test.tsx
Comment (studioswong) [inline components/header/header.test.tsx:38]: nit:
```suggestion
    //Open the menu
```

File: components/header/header.test.tsx
Comment (studioswong) [inline components/header/header.test.tsx:43]: nit:
```suggestion
    //Close the menu
```

File: components/header/header.module.scss
Comment (studioswong) [inline components/header/header.module.scss:11]: `rem` it please 😉

File: components/header/header.module.scss
Comment (studioswong) [reply components/header/header.module.scss:63]: not sure if it's just on my side (sometimes the gitUI does that), but I am seeing this as `px`?

**Review by studioswong — COMMENTED** [review]
I've tested the header, works well so far 😄 

Some follow up:

1. I noticed that the Kedro logo is a button but doesn't seem to be linked to anywhere - Is this supposed to link to our docs ? 
image

2. Please add in the links for the 'Get started' button.
3. As Tynan pointed, the documentation link needs to open in a seperate window.
4. In looking at the design, I noticed this other state with what seems to be a 'half' burger menu but don't seem to see this in the execution - I wonder what is this for? 

image

**Comment by rashidakanchwala** [comment]
> I've tested the header, works well so far 😄
> 
> Some follow up:
> 
> 1. I noticed that the Kedro logo is a button but doesn't seem to be linked to anywhere - Is this supposed to link to our docs ?
> 
> image
> 
> 2. Please add in the links for the 'Get started' button.
> 3. As Tynan pointed, the documentation link needs to open in a seperate window.
> 4. In looking at the design, I noticed this other state with what seems to be a 'half' burger menu but don't seem to see this in the execution - I wonder what is this for?
> 
> image



> 1. I noticed that the Kedro logo is a button but doesn't seem to be linked to anywhere - Is this supposed to link to our docs ?

1. this is when you go a section and then click on kedro logo to navigate back to the top of the page. 
2. Get Started section is still to be done hence no link also the comment to update links once section is ready.
4. it's just a slide show for Mackay shows the transition. I will bring it up in the standup and confirm with him.

File: components/header/header.tsx
Comment (rashidakanchwala) [reply components/header/header.tsx:21]: get started is to be updated once the section is ready. so i will leave it for now

File: components/hero/hero.module.scss
Comment (rashidakanchwala) [reply components/hero/hero.module.scss]: there were a lot of places where px was used. @tynandebold was ok for me to convert it to rem. So I have updated in this PR

**Comment by studioswong** [comment]
Makes sense, all clarified 😄

**Review by studioswong — APPROVED** [review]
good job for bringing this across 👍

**Pull request merged**
